Make `split_module` results deterministic

ansley opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#50470 Make `split_module` results deterministic**

Differential Revision: [D25899130](https://our.internmc.facebook.com/intern/diff/D25899130)

Commit:
Make `split_module` results deterministic

[ghstack-poisoned]

Commit:
Update on "Make `split_module` results deterministic"


Differential Revision: [D25899130](https://our.internmc.facebook.com/intern/diff/D25899130)

[ghstack-poisoned]

jamesr66a commented on torch/fx/experimental/subgraph_creation_example.py:
```diff
@@ -37,15 +37,15 @@ def record_cross_partition_use(def_node : torch.fx.node.Node, use_node : Optiona
             if def_partition_name is not None:
                 def_partition = partitions[def_partition_name]
-                def_partition.outputs.add(def_node.name)
+                def_partition.outputs[def_node.name] = None
```
Can also equivalently write `def_partition.outputs.setdefault(def_node.name)`

Commit:
Update on "Make `split_module` results deterministic"


Differential Revision: [D25899130](https://our.internmc.facebook.com/intern/diff/D25899130)

[ghstack-poisoned]

Commit:
Update on "Make `split_module` results deterministic"


Differential Revision: [D25899130](https://our.internmc.facebook.com/intern/diff/D25899130)

[ghstack-poisoned]
